### PR TITLE
feat(AddPeopleDialog): Update design; feat(SecurityDialog): Implement

### PR DESCRIPTION
**Pull request by muscat1**
Essentially, this PR
* Enables email invites to the conference, alongside a revamp of the `AddPeopleDialog`

* Implements a new Security dialog for the password feature, as well as future features

* Removes the info dialog in its entirety, splitting its components between the `AddPeopleDialog` and the `SecurityDialog`
* Adds a header when a user is alone in a conference, prompting them to invite others (also adding an emphasized toolbar button for the `SecurityDialog`)

Please take a look and also let me know if I've missed anything that should've been deleted

### CODE REVIEW

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.js [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+    }
+}
+
+/**
+ * Maps (parts of) the Redux state to the associated
+ * {@code AddPeopleDialog}'s props.
+ *
+ * @param {Object} state - The Redux state.
+ * @private
+ * @returns {{
```
Comment (zbettenbuk) [inline react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:462]: I suggest using `Props` as the return type here too

File: react/features/invite/components/add-people-dialog/web/LiveStreamSection.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                    <Icon src = { IconCheck } />
+                </>
+            );
+        }
+
+        return (
+            <>
+                <div
+                    className = { `invite-more-dialog stream-text${
+                        isHovered ? ' selected' : ''}`
```
Comment (zbettenbuk) [inline react/features/invite/components/add-people-dialog/web/LiveStreamSection.js:88]: Couldn't this be solved with pure CSS?

File: react/features/invite/components/add-people-dialog/web/utils.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+/**
+ * Tries to copy a given text to the clipboard.
+ *
+ * @param {string} textToCopy - Text to be copied.
+ * @returns {boolean}
+ */
+export function copyText(textToCopy: string) {
+    const fakeTextArea = document.createElement('textarea');
+
+    // $FlowFixMe
```
Comment (zbettenbuk) [inline react/features/invite/components/add-people-dialog/web/utils.js:17]: What was the flow error here?

File: react/features/invite/components/add-people-dialog/web/utils.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    document.body.appendChild(fakeTextArea);
```
Comment (zbettenbuk) [inline react/features/invite/components/add-people-dialog/web/utils.js:18]: Won't this cause intermittent flashing of the layout? You seem to append a visible element to the body. I think it's wise to at least make sure with some CSS that it won't make layout changes, even for a very brief moment.

File: interface_config.js [hunk 5]
```diff
@@ -52,7 +52,7 @@ var interfaceConfig = {
         'livestreaming', 'etherpad', 'sharedvideo', 'settings', 'raisehand',
         'videoquality', 'filmstrip', 'invite', 'feedback', 'stats', 'shortcuts',
         'tileview', 'videobackgroundblur', 'download', 'help', 'mute-everyone',
-        'e2ee'
+        'e2ee', 'security'
```
Comment (zbettenbuk) [inline interface_config.js:55]: Maybe i'm wrong, but you seem to removed the InfoDialogButton. If i'm right, then it would be nice to remove the related `info` button from here too.

File: react/features/toolbox/components/web/Toolbox.js [hunk 6]
```diff
@@ -1237,6 +1258,12 @@ class Toolbox extends Component<Props, State> {
         overflowMenuContent.splice(
             1, 0, ...this._renderMovedButtons(movedButtons));
 
+        const securityToolbarButtonStyles = {
```
Comment (zbettenbuk) [inline react/features/toolbox/components/web/Toolbox.js:1261]: Why did you need this inline style? Can't it be in a css file?

File: react/features/invite/components/security-dialog/PasswordSection.js [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+    }
+
+    /**
+     * Method to remotely submit the password from outside of the password form.
+     *
+     * @private
+     * @returns {void}
+     */
+    function onPasswordSave() {
+        // $FlowFixMe
```
Comment (zbettenbuk) [inline react/features/invite/components/security-dialog/PasswordSection.js:110]: This flow error seems valid. Can you please remove the exclusion and try to fix the error?

File: react/features/invite/components/security-dialog/PasswordSection.js [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
+    /**
+     * The number of digits to be used in the password.
+     */
+    passwordNumberOfDigits: ?number,
+
+    /**
+     * Action that sets the conference password.
+     */
+    setPassword: Function,
```
Comment (zbettenbuk) [inline react/features/invite/components/security-dialog/PasswordSection.js:46]: General observation: It's not very good to pass these kind of functions too deep down in the react dom tree, which seems to be happening here. Can't these props be constructed in the component where they are being used? E.g. by using `mapStateToProps` or `mapDispatchToProps`?

This concern applies to most other props of this component too.

File: react/features/invite/components/security-dialog/PasswordSection.js [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+     */
+    function onPasswordSave() {
+        // $FlowFixMe
+        formRef.current.querySelector('form').requestSubmit();
+    }
+
+    /**
+     * Callback invoked to unlock the current JitsiConference.
+     *
+     * @private
```
Comment (zbettenbuk) [inline react/features/invite/components/security-dialog/PasswordSection.js:117]: I generally don't like to use these `@private` flags (I think I actually never use them), because I deny the existence of private functions in JS (it's not TS, unfortunately). But if you like to use them, it's fine for me, just be consistent - e.g. this one is private, but the next two are not, whereas I think they are of the same theoretical visibility.

File: react/features/invite/components/security-dialog/PasswordSection.js [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+        );
+    }
+
+    return (
+        <div className = 'security-dialog password'>
+            <div
+                className = 'info-dialog info-dialog-column info-dialog-password'
+
+                // $FlowFixMe
+                ref = { formRef }>
```
Comment (zbettenbuk) [inline react/features/invite/components/security-dialog/PasswordSection.js:173]: Yep, here too. Please fix the flow error instead.

File: react/features/invite/components/security-dialog/SecurityDialog.js [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+ * Maps (parts of) the Redux state to the associated props for the
+ * {@code SecurityDialog} component.
+ *
+ * @param {Object} state - The Redux state.
+ * @private
+ * @returns {Props}
+ */
+function mapStateToProps(state) {
+    const {
+        conference,
```
Comment (zbettenbuk) [inline react/features/invite/components/security-dialog/SecurityDialog.js:108]: Props that are coming from Redux should have the `_` prefix

File: react/features/toolbox/components/web/ToolbarButton.js [hunk 12]
```diff
@@ -38,6 +38,7 @@ class ToolbarButton extends AbstractToolbarButton<Props> {
      * @static
      */
     static defaultProps = {
+        style: undefined,
```
Comment (zbettenbuk) [inline react/features/toolbox/components/web/ToolbarButton.js:41]: Do you need an undefined default value?

File: react/features/toolbox/components/web/Toolbox.js [hunk 13]
```diff
@@ -1093,13 +1108,20 @@ class Toolbox extends Component<Props, State> {
...
+                        icon = { this.props._locked ? IconLockPassword : IconUnlockPassword }
+                        key = 'security'
+                        onClick = { this._onToolbarOpenSecurity }
+                        text = { t('toolbar.security') } />
+                );
             case 'invite':
                 return (
                     <OverflowMenuItem
                         accessibilityLabel = { t('toolbar.accessibilityLabel.invite') }
-                        icon = { IconInvite }
```
Comment (zbettenbuk) [inline react/features/toolbox/components/web/Toolbox.js:1102]: I may be wrong, but I don't see this icon used anywhere else. If it's true, can you please remove the actual icon too?

File: react/features/invite/components/security-dialog/index.js [hunk 14]
```diff
@@ -0,0 +1 @@
+export { default as SecurityDialog } from './SecurityDialog';
```
Comment (zbettenbuk) [inline react/features/invite/components/security-dialog/index.js:1]: Can you please make sure you add `// @flow` to every file? Even index files, it's very important.

File: react/features/conference/components/web/Conference.js [hunk 15]
```diff
@@ -202,6 +203,7 @@ class Conference extends AbstractConference<Props, *> {
 
                 <Notice />
                 <Subject />
+                <InviteMore />
```
[the remark this answers is not in the document]
Comment (muscat1) [reply react/features/conference/components/web/Conference.js:206]: This is actually the subject replacement when there's a single participant in the conference, the modal is invoked via the `beginAddPeople` action.

File: react/features/invite/components/add-people-dialog/web/DialInNumber.js [hunk 16]
```diff
@@ -44,12 +46,16 @@ class DialInNumber extends Component<Props> {
...
+        const pinLabel = t('info.dialInConferenceID');
+        const conferenceIDPin = `${_formatConferenceIDPin(conferenceID)}#`;
+        const textToCopy = `${dialInLabel} ${phoneNumber} ${pinLabel} ${conferenceIDPin}`;
 
         return (
             <div className = 'dial-in-number'>
                 <span className = 'phone-number'>
                     <span className = 'info-label'>
-                        { t('info.dialInNumber') }
+                        { dialInLabel }
```
[the remark this answers is not in the document]
Comment (muscat1) [reply react/features/invite/components/add-people-dialog/web/DialInNumber.js:58]: It was in order to avoid calling the translation function again for the `textToCopy` but I'll be extracting them in a separate click handler

File: react/features/invite/components/add-people-dialog/web/DialInSection.js [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+};
+
+/**
+ * Returns a ReactElement for showing how to dial into the conference, if
+ * dialing in is available.
+ *
+ * @private
+ * @returns {null|ReactElement}
+ */
+function DialInSection({
```
[the remark this answers is not in the document]
Comment (muscat1) [reply react/features/invite/components/add-people-dialog/web/DialInSection.js:45]: :D

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.js [hunk 18]
```diff
@@ -0,0 +1,489 @@
...
+     * @param {Object} prevProps - The state object before the update.
+     * @param {Object} prevState - The state object before the update.
+     * @returns {void}
+     */
+    componentDidUpdate(prevProps, prevState) {
+        /**
+         * Clears selected items from the multi select component on successful
+         * invite.
+         */
+        if (prevState.addToCallError
```
[the remark this answers is not in the document]
Comment (muscat1) [reply react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:93]: This is the code that was in the original `AddPeopleDialog` component, without the `Dialog` wrapper. I'm guessing this is looking at the `_multiselect` prop to change.

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.js [hunk 19]
```diff
@@ -0,0 +1,489 @@
...
+     * @returns {ReactElement}
+     */
+    render() {
+        const {
+            _addPeopleEnabled,
+            _dialOutEnabled,
+            _footerTextEnabled,
+            t
+        } = this.props;
+        let isMultiSelectDisabled = this.state.addToCallInProgress || false;
```
[the remark this answers is not in the document]
Comment (muscat1) [reply react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:113]: No idea :D This was in the original (I guess legacy, code) but it does seem redundant, now that you mention it.

File: react/features/invite/components/add-people-dialog/web/utils.js (hunk 3 above)
Comment (muscat1) [reply react/features/invite/components/add-people-dialog/web/utils.js:17]: IIRC it implied that the `document.body` could be null and that seemed like a weird error to me, hence the `$FlowFixMe`. I could add a null check, but it seems redundant to me..

File: react/features/invite/components/security-dialog/PasswordSection.js (hunk 9 above)
Comment (muscat1) [reply react/features/invite/components/security-dialog/PasswordSection.js:117]: I actually agree with your opinion, sometimes copy-pasta goes wrong though...

File: react/features/invite/components/security-dialog/PasswordSection.js (hunk 8 above)
Comment (muscat1) [reply react/features/invite/components/security-dialog/PasswordSection.js:46]: I was trying to achieve a controlled component type of implementation, where the dialog is the parent. If you're adamant about this, I could change it, no problem

File: lang/main.json [hunk 20]
```diff
@@ -1,21 +1,36 @@
 {
     "addPeople": {
         "add": "Invite",
+        "addContacts": "Invite your contacts",
+        "copyInvite": "Copy meeting invite",
```
Comment (saghul) [inline lang/main.json:5]:
```suggestion
        "copyInvite": "Copy meeting invitation",
```

File: lang/main.json [hunk 21]
```diff
@@ -1,21 +1,36 @@
...
         "loading": "Searching for people and phone numbers",
         "loadingNumber": "Validating phone number",
         "loadingPeople": "Searching for people to invite",
         "noResults": "No matching search results",
         "noValidNumbers": "Please enter a phone number",
+        "outlookEmail": "Outlook Email",
         "searchNumbers": "Add phone numbers",
         "searchPeople": "Search for people",
         "searchPeopleAndNumbers": "Search for people or add their phone numbers",
+        "shareInvite": "Share meeting invite",
```
Comment (saghul) [inline lang/main.json:28]:
```suggestion
        "shareInvite": "Share meeting invitation",
```

File: lang/main.json [hunk 22]
```diff
@@ -1,21 +1,36 @@
...
+        "shareLink": "Share the meeting link to invite others",
```
Comment (saghul) [inline lang/main.json:29]: What is this?

File: react/features/base/dialog/components/web/StatelessDialog.js [hunk 23]
```diff
@@ -116,8 +122,11 @@ class StatelessDialog extends Component<Props> {
         return (
             <Modal
                 autoFocus = { true }
+                components = {{
+                    Header: CustomHeader ?? undefined
```
Comment (saghul) [inline react/features/base/dialog/components/web/StatelessDialog.js:126]: looks like having default props would help here

File: react/features/conference/components/web/InviteMore.js [hunk 24]
```diff
@@ -0,0 +1,88 @@
...
+    return {
+        isVisible: isToolboxVisible(state) && participantCount === 1
+    };
+}
+
+const mapDispatchToProps = {
+    onClick: () => beginAddPeople()
+};
+
+export default translate(
```
Comment (saghul) [inline react/features/conference/components/web/InviteMore.js:83]: this shouldd fit in one line (character limit is 120, remember)

File: react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js [hunk 25]
```diff
@@ -1,478 +1,209 @@
...
+    updateNumbers: Function
+};
 
-    _onSelectionChange: (Map<*, *>) => void;
+/**
+ * Invite More component.
+ *
+ * @returns {React$Element<any>}
+ */
+function AddPeopleDialog({
```
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js:79]: Looks less complex because he used hooks. I'm not a fan (yet), but let's give it a try!

File: react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js [hunk 26]
```diff
@@ -1,478 +1,209 @@
...
+    const [ phoneNumber, setPhoneNumber ] = useState(undefined);
+
+    /**
+     * Updates the dial-in numbers.
+     */
+    useEffect(() => {
+        if (!dialIn.numbers) {
+            updateNumbers();
+        }
+    }, []);
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js:98]: This effect should filter by the dialIn numbers, right?

File: react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js [hunk 27]
```diff
@@ -1,478 +1,209 @@
...
-     * @private
      * @returns {void}
      */
-    _onSelectionChange(selectedItems) {
-        this.setState({
-            inviteItems: selectedItems
-        });
-    }
+    useEffect(() => {
+        sendAnalytics(createInviteDialogEvent(
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js:106]: This wouldd be triggered on every render, we probably don't want to do that

File: react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js [hunk 28]
```diff
@@ -1,478 +1,209 @@
...
+            'invite.dialog.opened', 'dialog'));
 
-    _onSubmit: () => void;
+        return () => {
+            sendAnalytics(createInviteDialogEvent(
+                'invite.dialog.closed', 'dialog'));
+        };
+    }, []);
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js:113]: remocve the filter if we are not applying it

File: react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js [hunk 29]
```diff
@@ -1,478 +1,209 @@
...
+        phoneNumber,
+        t
+    });
+    const inviteSubject = t('addPeople.inviteMoreMailSubject', {
+        appName: interfaceConfig.APP_NAME
+    });
+
+    return (
+        <Dialog
+            CustomHeader = { Header }
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js:143]: camel case this prop please

File: react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js [hunk 30]
```diff
@@ -1,478 +1,209 @@
...
-
-    if (enableFeaturesBasedOnToken) {
-        const { features = {} } = getLocalParticipant(state);
-
-        if (String(features['outbound-call']) !== 'true') {
-            footerTextEnabled = true;
-        }
-    }
+function mapStateToProps(state) {
+    const { room } = state['features/base/conference'];
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js:181]: IIRC we have a getRoomName getter, please check it out.

File: react/features/invite/components/add-people-dialog/web/DialInNumber.js [hunk 31]
```diff
@@ -44,12 +46,16 @@ class DialInNumber extends Component<Props> {
      */
     render() {
         const { conferenceID, phoneNumber, t } = this.props;
+        const dialInLabel = t('info.dialInNumber');
+        const pinLabel = t('info.dialInConferenceID');
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/DialInNumber.js:50]: Can you rename this to passcode?

File: react/features/invite/components/add-people-dialog/web/DialInSection.js [hunk 32]
```diff
@@ -0,0 +1,73 @@
+// @flow
+import React from 'react';
+
+import DialInNumber from './DialInNumber';
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/DialInSection.js:4]: style: this one should go lasst because it's the shallowest path

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.js [hunk 33]
```diff
@@ -0,0 +1,489 @@
...
+import { MultiSelectAutocomplete } from '../../../../base/react';
+import { translate, translateToHTML } from '../../../../base/i18n';
+
+declare var interfaceConfig: Object;
+
+type Props = AbstractProps & {
+
+    /**
+     * The {@link JitsiMeetConference} which will be used to invite "room"
+     * participants through the SIP Jibri (Video SIP gateway).
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:24]: It's actually Jigasi. No need to mention how we invite them, since this may change in the future.

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:93]: But multiselect is not a prop...

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.js [hunk 34]
```diff
@@ -0,0 +1,489 @@
...
+     * @private
+     * @returns {Object[]} Configuration objects for items to display in the
+     * search autocomplete.
+     */
+    _parseQueryResults(response = []) {
+        const { t, _dialOutEnabled } = this.props;
+        const users = response.filter(item => item.type !== 'phone');
+        const userDisplayItems = [];
+
+        users.forEach(user => {
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:273]: use `for..of` please.

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.js [hunk 35]
```diff
@@ -0,0 +1,489 @@
...
+     * @returns {ReactElement|null}
+     */
+    _renderErrorMessage() {
+        if (!this.state.addToCallError) {
+            return null;
+        }
+
+        const { t } = this.props;
+        const supportString = t('inlineDialogFailure.supportMsg');
+        const supportLink = interfaceConfig.SUPPORT_URL;
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:397]: check if it's undefined?

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.js [hunk 36]
```diff
@@ -0,0 +1,489 @@
...
+ *     _dialOutAuthUrl: string,
+ *     _jwt: string,
+ *     _peopleSearchQueryTypes: Array<string>,
+ *     _peopleSearchUrl: string
+ * }}
+ */
+function _mapStateToProps(state) {
+    const {
+        enableFeaturesBasedOnToken
+    } = state['features/base/config'];
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:472]: this should fit on one line

File: react/features/invite/components/add-people-dialog/web/utils.js [hunk 37]
```diff
@@ -0,0 +1,80 @@
...
+
+    return result;
+}
+
+/**
+ * Creates a message describing how to dial in to the conference.
+ *
+ * @returns {string}
+ */
+export function getInviteText({
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/utils.js:35]: Mobile could reuse this, maybe place it in some common file?

File: react/features/invite/components/index.js [hunk 38]
```diff
@@ -2,5 +2,5 @@
 
 export * from './add-people-dialog';
 export * from './dial-in-summary';
-export * from './info-dialog';
 export * from './callee-info';
+export * from './security-dialog';
```
Comment (saghul) [inline react/features/invite/components/index.js:6]: The security dialog is not related with inviting people, is it? Then let's move it some other feature.

File: react/features/invite/components/security-dialog/PasswordSection.js [hunk 39]
```diff
@@ -0,0 +1,188 @@
...
+    /**
+     * Callback invoked to set a password on the current JitsiConference.
+     *
+     * @param {string} enteredPassword - The new password to be used to lock the
+     * current JitsiConference.
+     * @private
+     * @returns {void}
+     */
+    function onPasswordSubmit(enteredPassword) {
+        setPassword(
```
Comment (saghul) [inline react/features/invite/components/security-dialog/PasswordSection.js:85]: doesn't this fit in a singlee line?

File: react/features/toolbox/components/web/Toolbox.js [hunk 40]
```diff
@@ -1093,13 +1108,20 @@ class Toolbox extends Component<Props, State> {
                 );
             case 'closedcaptions':
                 return <ClosedCaptionButton showLabel = { true } />;
-            case 'info':
-                return <InfoDialogButton showLabel = { true } />;
+            case 'security':
+                return (
+                    <OverflowMenuItem
```
Comment (saghul) [inline react/features/toolbox/components/web/Toolbox.js:1113]: Can you please ccreate a button instead?

File: react/features/toolbox/components/web/Toolbox.js [hunk 41]
```diff
@@ -1283,13 +1310,18 @@ class Toolbox extends Component<Props, State> {
...
-                            icon = { IconInvite }
+                            icon = { IconInviteMore }
                             onClick = { this._onToolbarOpenInvite }
                             tooltip = { t('toolbar.invite') } /> }
-                    {
-                        buttonsRight.indexOf('info') !== -1
-                            && <InfoDialogButton />
-                    }
+                    { buttonsRight.indexOf('security') !== -1
+                        && <ToolbarButton
```
Comment (saghul) [inline react/features/toolbox/components/web/Toolbox.js:1317]: Please create a button which is based of AbstractButton, then you can use the same component on the toolbar or thee overflow menu.

File: react/features/toolbox/components/web/Toolbox.js [hunk 42]
```diff
@@ -1093,13 +1108,20 @@ class Toolbox extends Component<Props, State> {
                 );
             case 'closedcaptions':
                 return <ClosedCaptionButton showLabel = { true } />;
-            case 'info':
-                return <InfoDialogButton showLabel = { true } />;
+            case 'security':
+                return (
+                    <OverflowMenuItem
+                        accessibilityLabel = { t('toolbar.accessibilityLabel.security') }
+                        icon = { this.props._locked ? IconLockPassword : IconUnlockPassword }
```
Comment (saghul) [inline react/features/toolbox/components/web/Toolbox.js:1115]: when you do a button, this should be taken care of within it.

File: lang/main.json (hunk 22 above)
Comment (muscat1) [reply lang/main.json:29]: The text above the `CopyLinkSection` component that prompts users to copy the conference URL

File: react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js (hunk 26 above)
Comment (muscat1) [reply react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js:98]: Not in this specific case, this used to be called when the `InviteToolbarButton` mounted, so I moved it here, when the actual dialog mounts.

File: react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js (hunk 27 above)
Comment (muscat1) [reply react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js:106]: To my knowledge, by passing the `[]` as a second arg to the effect, it only runs it once per component lifecycle, acting as a sort of `componentDidMount` and `componentWillUnmount` in one (via the returned cleanup function).

File: react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js (hunk 28 above)
Comment (muscat1) [reply react/features/invite/components/add-people-dialog/web/AddPeopleDialog.js:113]: I'm not quite sure what you mean by filter, if it's about the 2nd argument of the `setEffect`, it's necessary to ensure the effect and its cleanup function are only run once.

**Comment by saghul** [comment]
Jenkins please test this please.

File: react/features/conference/components/web/Conference.js [hunk 43]
```diff
@@ -33,6 +33,7 @@ import {
     abstractMapStateToProps
 } from '../AbstractConference';
 import type { AbstractProps } from '../AbstractConference';
+import InviteMore from './InviteMore';
```
[the remark this answers is not in the document]
Comment (muscat1) [reply react/features/conference/components/web/Conference.js:34]: Noted, thanks!

File: interface_config.js (hunk 5 above)
Comment (muscat1) [reply interface_config.js:55]: Yeah you're right, removed it

**Comment by saghul** [comment]
Jenkins please test this please.

File: react/features/invite/components/add-people-dialog/web/utils.js (hunk 4 above)
Comment (muscat1) [reply react/features/invite/components/add-people-dialog/web/utils.js:13]: Not at all from what I've tested, most likely due to the current layout.

**Comment by saghul** [comment]
Jenkins please test this please.

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (muscat1) [reply react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:95]: Still not too sure about this, but again, it was part of the codebase that I didn't interfere with. If anything acts up, I'll come back to it

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.js (hunk 35 above)
Comment (muscat1) [reply react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:412]: Would that be needed? The support url is hardcoded in interface_config.js atm. If it could indeed be undefined, what should we default to?

Comment (saghul) [reply react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:412]: Maybe nothing? Just ddon't show anything.

Comment (saghul) [reply react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:412]: Maybe it shouldn't have a default value, since people running their own installs would have a link to us. Anyway, food for thought.

**Comment by saghul** [comment]
Jenkins please test this please.

File: lang/main.json [hunk 44]
```diff
@@ -1,21 +1,36 @@
 {
     "addPeople": {
         "add": "Invite",
+        "addContacts": "Invite your contacts",
```
Comment (saghul) [inline lang/main.json:4]: General comment about this file. We need to get rid of the info section and put things in the right place. Let's do that in a follow-up PR.

File: css/modals/invite/_info.scss [hunk 45]
```diff
@@ -3,47 +3,6 @@
     display: flex;
     font-size: 14px;
 
-    .info-dialog-action-link {
```
Comment (saghul) [inline css/modals/invite/_info.scss:6]: This entire file should disappear. Let's do that in a follow-up PR.

File: react/features/base/dialog/components/web/StatelessDialog.js [hunk 46]
```diff
@@ -31,6 +31,11 @@ type Props = {
 
     i18n: Object,
 
+    /**
+     * Custom dialog header that replaces the standard heading.
+     */
+    customHeader?: React$Element<any> | Function,
```
Comment (saghul) [inline react/features/base/dialog/components/web/StatelessDialog.js:37]: alpha sort!

File: react/features/base/dialog/components/web/StatelessDialog.js [hunk 47]
```diff
@@ -116,8 +122,11 @@ class StatelessDialog extends Component<Props> {
         return (
             <Modal
                 autoFocus = { true }
+                components = {{
+                    Header: customHeader ?? undefined
```
Comment (saghul) [inline react/features/base/dialog/components/web/StatelessDialog.js:126]: do you really need to do this? if the prop is not given it will be undefined...

File: react/features/conference/components/web/InviteMore.js [hunk 48]
```diff
@@ -0,0 +1,93 @@
...
+ */
+function InviteMore({
+    _tileViewEnabled,
+    _visible,
+    onClick,
+    t
+}: Props) {
+    return (
+        _visible
+            ? <div className = { `invite-more-container${_tileViewEnabled ? ' elevated' : ''}` }>
```
Comment (saghul) [inline react/features/conference/components/web/InviteMore.js:50]: is there a problem if we always apply the elevated class?

File: react/features/invite/components/add-people-dialog/web/InviteByEmailSection.js [hunk 49]
```diff
@@ -0,0 +1,154 @@
...
+
+    /**
+     * Opens an email provider containing the conference invite.
+     *
+     * @param {string} url - The url to be opened.
+     * @returns {Function}
+     */
+    function _onSelectProvider(url) {
+        return function() {
+            window.open(url, '_blank');
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/InviteByEmailSection.js:63]: `openURLInBrowser` from base/util

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.js [hunk 50]
```diff
@@ -0,0 +1,495 @@
...
+     */
+    _renderErrorMessage() {
+        if (!this.state.addToCallError) {
+            return null;
+        }
+
+        const { t } = this.props;
+        const supportString = t('inlineDialogFailure.supportMsg');
+        const supportLink = interfaceConfig.SUPPORT_URL;
+        const supportLinkContent = supportLink && (
```
Comment (saghul) [inline react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:412]: here you'd still be rendering the inline message below, I think we'd need to skip that too.

File: react/features/security/components/security-dialog/PasswordForm.js [hunk 51]
```diff
@@ -1,9 +1,8 @@
 // @flow
-
 import React, { Component } from 'react';
 
-import { translate } from '../../../../base/i18n';
-import { LOCKED_LOCALLY } from '../../../../room-lock';
+import { LOCKED_LOCALLY } from '../../../room-lock';
+import { translate } from '../../../base/i18n';
```
Comment (saghul) [inline react/features/security/components/security-dialog/PasswordForm.js:5]: alpha sort

File: react/features/security/components/security-dialog/SecurityDialog.js [hunk 52]
```diff
@@ -0,0 +1,124 @@
+// @flow
+import React, { useState, useEffect } from 'react';
+
+import { connect } from '../../../base/redux';
+import { Dialog } from '../../../base/dialog';
+import Header from './Header';
+import { isLocalParticipantModerator } from '../../../base/participants';
+import PasswordSection from './PasswordSection';
+import { setPassword as setPass } from '../../../base/conference';
```
Comment (saghul) [inline react/features/security/components/security-dialog/SecurityDialog.js:9]: alpha sort

File: react/features/security/components/security-dialog/SecurityDialogButton.js [hunk 53]
```diff
@@ -0,0 +1,82 @@
+// @flow
+import { AbstractButton, type AbstractButtonProps } from '../../../base/toolbox';
```
Comment (saghul) [inline react/features/security/components/security-dialog/SecurityDialogButton.js:2]: ditto

also, please leave an empty line after `// @flow`

File: react/features/security/components/security-dialog/SecurityDialogButton.js [hunk 54]
```diff
@@ -0,0 +1,82 @@
...
+     */
+    onClick: Function
+
+};
+
+
+/**
+ * Implements an {@link AbstractButton} to open the security dialog.
+ */
+class SharedDocumentButton extends AbstractButton<Props, *> {
```
Comment (saghul) [inline react/features/security/components/security-dialog/SecurityDialogButton.js:29]: SharedDocument?!

File: react/features/toolbox/components/web/Toolbox.js [hunk 55]
```diff
@@ -36,12 +36,8 @@ import { ChatCounter, toggleChat } from '../../../chat';
...
 import { SharedDocumentButton } from '../../../etherpad';
 import { openFeedbackDialog } from '../../../feedback';
-import {
-    beginAddPeople,
-    InfoDialogButton,
-    isAddPeopleEnabled,
-    isDialOutEnabled
-} from '../../../invite';
+import { beginAddPeople } from '../../../invite';
+import { SecurityDialogButton } from '../../../security';
```
Comment (saghul) [inline react/features/toolbox/components/web/Toolbox.js:40]: alpha sort

File: react/features/base/dialog/components/web/StatelessDialog.js (hunk 47 above)
Comment (muscat1) [reply react/features/base/dialog/components/web/StatelessDialog.js:126]: IIRC when I was testing I needed to explicitly pass that but I could be mistaken, will double check and alter if that's not the case

File: react/features/invite/components/add-people-dialog/web/InviteContactsForm.js (hunk 50 above)
Comment (muscat1) [reply react/features/invite/components/add-people-dialog/web/InviteContactsForm.js:412]: Oh damn, you're right

File: react/features/security/components/security-dialog/SecurityDialogButton.js (hunk 54 above)
Comment (muscat1) [reply react/features/security/components/security-dialog/SecurityDialogButton.js:29]: ... Ehm... yeah, ignore that :P

File: react/features/conference/components/web/InviteMore.js (hunk 48 above)
Comment (muscat1) [reply react/features/conference/components/web/InviteMore.js:50]: Yeah, in large view the header will cover the (me) preview tile, due to its lower z-index.

Comment (saghul) [reply react/features/conference/components/web/InviteMore.js:50]: You mean the header covers the entire large view and not just a top portion of it?

File: react/features/toolbox/components/web/Toolbox.js [hunk 56]
```diff
@@ -1283,13 +1283,10 @@ class Toolbox extends Component<Props, State> {
...
                                 { t('toolbar.accessibilityLabel.invite') }
-                            icon = { IconInvite }
+                            icon = { IconInviteMore }
                             onClick = { this._onToolbarOpenInvite }
                             tooltip = { t('toolbar.invite') } /> }
-                    {
-                        buttonsRight.indexOf('info') !== -1
-                            && <InfoDialogButton />
-                    }
+                    <SecurityDialogButton customClass = 'security-toolbar-button' />
```
Comment (saghul) [inline react/features/toolbox/components/web/Toolbox.js:1289]: re-add the checck now please, for both info / security

**Comment by saghul** [comment]
Jenkins please test this please.

**Comment by saghul** [comment]
(landed with GH action failing because GH actions sseems to have an outage, Jenkins is fine)

**Comment by luixxiul** [comment]
Kudos to the implementation :shipit:
